ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ResponseMapper.java

```diff
@@ -0,0 +1,274 @@
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.isAllBlank;
+import static uk.gov.hmcts.cmc.ccd.util.StreamUtil.asStream;
+import static uk.gov.hmcts.cmc.domain.models.response.YesNoOption.NO;
+
+@Component
+public class ResponseMapper {
+    private final EvidenceRowMapper evidenceRowMapper;
+    private final TimelineEventMapper timelineEventMapper;
+    private final DefendantPartyMapper defendantPartyMapper;
+    private final PaymentIntentionMapper paymentIntentionMapper;
+    private final StatementOfMeansMapper statementOfMeansMapper;
+
+    public ResponseMapper(
+        EvidenceRowMapper evidenceRowMapper,
+        TimelineEventMapper timelineEventMapper,
+        DefendantPartyMapper defendantPartyMapper,
+        PaymentIntentionMapper paymentIntentionMapper,
+        StatementOfMeansMapper statementOfMeansMapper
+    ) {
+        this.evidenceRowMapper = evidenceRowMapper;
+        this.timelineEventMapper = timelineEventMapper;
+        this.defendantPartyMapper = defendantPartyMapper;
+        this.paymentIntentionMapper = paymentIntentionMapper;
+        this.statementOfMeansMapper = statementOfMeansMapper;
+    }
+
+    public void to(CCDDefendant.CCDDefendantBuilder builder, Response response) {
+        requireNonNull(builder, "builder must not be null");
+        requireNonNull(response, "response must not be null");
+
+        builder.responseType(
+            CCDResponseType.valueOf(response.getResponseType().name())
+        );
+
+        builder.responseFreeMediationOption(
+            CCDYesNoOption.valueOf(response.getFreeMediation().orElse(NO).name())
+        );
+
+        if (response.getMoreTimeNeeded() != null) {
+            builder.responseMoreTimeNeededOption(CCDYesNoOption.valueOf(response.getMoreTimeNeeded().name()));
+        }
+
+        response.getStatementOfTruth().ifPresent(
+            statementOfTruth -> {
+                builder.responseDefendantSOTSignerName(statementOfTruth.getSignerName());
+                builder.responseDefendantSOTSignerRole(statementOfTruth.getSignerRole());
+            }
+        );
+        defendantPartyMapper.to(builder, response.getDefendant());
+
+        switch (response.getResponseType()) {
+            case FULL_DEFENCE:
+                toFullDefenceResponse(builder, (FullDefenceResponse) response);
+                break;
+            case FULL_ADMISSION:
+                toFullAdmissionResponse(builder, (FullAdmissionResponse) response);
+                break;
+            case PART_ADMISSION:
+                toPartAdmissionResponse(builder, (PartAdmissionResponse) response);
+                break;
+            default:
+                throw new MappingException("Invalid response type " + response.getResponseType());
+        }
+    }
+
+    private void toPartAdmissionResponse(CCDDefendant.CCDDefendantBuilder builder, PartAdmissionResponse response) {
```

Comment:
can these methods be part of another class PartAdmissionResponseMapper? to keep the from and to method specific to what the class does. Also the class can be smaller than now being very big.

Reply:
Agree we can extract them out.